### ISSUE TITLE
Blend mode that increases saturation under areas of low value

### ISSUE THREAD

**Issue by danirabbit** [elementary/gala#946]
## Problem

Currently shadows overlay black. This only affects value but does not affect saturation as shadows in real life would:

![screen2](https://user-images.githubusercontent.com/7277719/97121542-4cadb200-16dc-11eb-8e37-eb9f082cdb07.png)

## Proposal

It would be pretty sweet if we could increase saturation under shadows so that they would be tinted with the color of what's beneath them and look more true to life:

![screen1](https://user-images.githubusercontent.com/7277719/97121562-6e0e9e00-16dc-11eb-939d-02b0b756e454.png)

**Comment by jassmith** [comment]
I made you a dumb example doing this with 2 shadows and simple blending modes in CSS. Not sure you really need a custom shader. Its definitely not flawless but it might be a simple place to start.

https://codepen.io/jassmith87/pen/ExyWgav

**Comment by donadigo** [comment]
I don't think this is a feasible thing to do in gala at least partially. Shadows in Gtk3 are being drawn by the client itself, not the WM.
If we were to do it just here in gala we would need to:
1. Render the window to some offscreen framebuffer first.
2. Either:
* Render our own shadow first. Then render just the window area of the offscreen buffer to the screen (if rectangular it would leave artrifacts on windows with rounded edges).
* Apply & composite a shader to the buffer with a mask, which still has the same issue with rounded corners, unless we specifically make the assumption that the rounded corners are always present in all windows.

Both methods assume a lot about how a window looks like but this cannot be fixed either in mutter, because of the chosen architecture by Gtk.

**Comment by hanaral** [comment]
> I don't think this is a feasible thing to do in gala at least partially. Shadows in Gtk3 are being drawn by the client itself, not the WM.
> If we were to do it just here in gala we would need to:
> 
>     1. Render the window to some offscreen framebuffer first.
> 
>     2. Either:
> 
> 
>     * Render our own shadow first. Then render just the window area of the offscreen buffer to the screen (if rectangular it would leave artrifacts on windows with rounded edges).
> 
>     * Apply & composite a shader to the buffer with a mask, which still has the same issue with rounded corners, unless we specifically make the assumption that the rounded corners are always present in all windows.
> 
> 
> Both methods assume a lot about how a window looks like but this cannot be fixed either in mutter, because of the chosen architecture by Gtk.

I thought Gala used it's own shadows anyway, hence the odd incompatabilites of the stylesheet with other WMs. (I haven't tried it with the new stylesheet though)

**Comment by cassidyjames** [comment]
>I thought Gala used it's own shadows anyway

No, shadows are client-side in the GTK stylesheet, and rendered by GTK.

**Comment by lenemter** [comment]
Closing because there's no way to implement this
